vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -1028,6 +1048,27 @@ func TestAccResourceVSphereVirtualMachine(t *testing.T) {
 				},
 			},
 		},
+		{
+			"clone into empty cluster, no environment browser (fails on plan)",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+					testAccResourceVSphereVirtualMachinePreCheck(tp)
+				},
+				Providers: testAccProviders,
+				Steps: []resource.TestStep{
+					{
+						Config:      testAccResourceVSphereVirtualMachineConfigCloneEmptyCluster(),
+						ExpectError: regexp.MustCompile("compute resource .* is missing an Environment Browser\\. Check host, cluster, and vSphere license health of all associated resources and try again"),
+						PlanOnly:    true,
+					},
+					{
+						Config: testAccResourceVSphereEmpty,
+						Check:  resource.ComposeTestCheckFunc(),
```

Comment:
Is this really supposed to execute the config and not check anything?

Reply:
This is necessary for steps with `PlanOnly` to ensure that the destroy step does not attempt to plan/refresh off of the broken configuration and run into the same issue. I think this might be a bug in the test framework but I'm not 100% sure, as the destroy step is only supposed to run if a state exists as referenced [here](https://github.com/hashicorp/terraform/blob/master/helper/resource/testing.go#L566). and it's odd to me as to why a bad plan would have a state versus a bad apply. In any case, the second step just runs with a completely blank config to clear out the state so any subsequent destroy steps have nothing to do.

An example of what happens when that step does not run:

```
    --- FAIL: TestAccResourceVSphereVirtualMachine/clone_into_empty_cluster,_no_environment_browser_(fails_on_plan) (5.54s)
    	testing.go:573: Error destroying resource! WARNING: Dangling resources
    		may exist. The full state and error is shown below.
    		
    		Error: Error refreshing: 1 error(s) occurred:
    		
    		* vsphere_virtual_machine.vm: 1 error(s) occurred:
    		
    		* vsphere_virtual_machine.vm: cannot find OS family for guest ID "ubuntu64Guest": compute resource "ComputeResource:domain-c178 @ /hashi-dc/host/hashi-cluster2" is missing an Environment Browser. Check host, cluster, and vSphere license health of all associated resources and try again
    		
    		State: <nil>
```

I've removed it from the apply test (creation from scratch) and will push.